src/mmap.rs

```diff
@@ -0,0 +1,89 @@
+use std::fs::File;
+use std::io::Error;
+use std::io::ErrorKind;
+use std::ops::Deref;
+use std::os::unix::io::AsRawFd;
+use std::ptr::null_mut;
+use std::slice;
+
+#[derive(Debug)]
+pub(crate) struct Mmap {
+    ptr: *mut libc::c_void,
+    len: usize,
+}
+
+impl Mmap {
+    pub fn map(file: &File) -> Result<Self, Error> {
+        let len = libc::size_t::try_from(file.metadata()?.len())
+            .map_err(|_err| Error::new(ErrorKind::InvalidData, "file is too large to mmap"))?;
+        let offset = 0;
+
+        // SAFETY: `mmap` with the provided arguments is always safe to call.
+        let ptr = unsafe {
+            libc::mmap(
+                null_mut(),
+                len,
+                libc::PROT_READ,
+                libc::MAP_SHARED,
+                file.as_raw_fd(),
+                offset,
```

Comment:
why a separate variable for passing 0 here, for documentation purposes?

Reply:
> why a separate variable for passing 0 here, for documentation purposes?

Yep, won't be obvious to the two-weeks-from-now-me what it represents otherwise.